src/tag.tsx

```diff
@@ -2,35 +2,95 @@ import * as React from 'react';
 import {useTheme} from './hooks';
 import {createUseStyles} from './jss';
 import {Text} from './text';
-import {ThemeVariant} from './theme-variant-context';
+import {ThemeVariant, useIsInverseVariant} from './theme-variant-context';
+import {pxToRem} from './utils/css';
 
 const useStyles = createUseStyles(() => ({
     tag: {
-        backgroundColor: ({color}) => color,
-        padding: '2px 8px',
-        borderRadius: 2,
-        minWidth: 48,
         display: 'inline-flex',
+        flexDirection: 'row',
         alignItems: 'center',
+        borderRadius: 50,
         justifyContent: 'center',
+        minWidth: 56,
+        padding: ({hasIcon}) => `4px 12px 4px ${hasIcon ? 8 : 12}px`,
+
+        // FIXME: remove this style and use an inline style for the icon once WEB-338 gets merged
+        '& svg': {
+            display: 'block',
+        },
     },
 }));
 
+type IconProps = {color?: string; size?: string | number};
+
+type TagType = 'promo' | 'active' | 'inactive' | 'success' | 'warning' | 'error';
+
 export type TagProps = {
+    // not using "TagType" and repeating the union to make these props playroom autocomplete friendly
+    type?: 'promo' | 'active' | 'inactive' | 'success' | 'warning' | 'error';
     children: string;
-    color: string;
+    Icon?: React.FC<IconProps>;
```

Comment:
why `Icon` prop is capitalized?
![image](https://user-images.githubusercontent.com/6722153/143249140-cf3802a3-f4e6-4f7b-8ffc-45dd5eaba9ce.png)

Reply:
Because it represents a component and components use PascalCase

You would use the Tag component as:
```js
<Tag Icon={IconPatata}>Patata</Tag>
```

instead of:
```js
<Tag icon={<IconPatata size="16" color="someColor" />}>Patata</Tag>
```

This way the Tag component is who decides how to render the icon (with a specific size and color)

Reply:
could be more intutive maybe... `iconName` ?

`<Tag iconName={IconPatataRegular}>Patata</Tag>`

Reply:
`iconName` is less intuitive, I would expect that prop to be a string.

A more obvious name would be `IconComponent` but I prefer the short version